ruff.toml

```diff
@@ -0,0 +1,83 @@
+# Enable pycodestyle (`E`) and Pyflakes (`F`) codes by default.
+select = [
+    # Pyflakes
+    "F",
+    # Pycodestyle
+    "E",
+    "W",
+    # flake8-docstrings
+    "D",
+    # isort
+    "I",
+]
+ignore = [
+    # TODO: Fix these.
+    # Lines longer than line-length. This is generally handled by Black.
+    "E501",
+    # Missing docstring in public module
+    "D100",
+    # Missing docstring in public package
+    "D104",
+    # no-blank-line-before-class
+    "D211",
+    # First line should end with a period
+    "D400",
+    # First word of the docstring should not be "This"
+    "D404",
+    # First line of docstring should be in imperative mood
+    "D401",
+    # Missing argument descriptions in the docstring
+    "D417",
+
+]
+
+# Allow autofix for all enabled rules (when `--fix`) is provided.
+fixable = [
+    "A", "B", "C", "D", "E", "F", "G", "I", "N", "Q", "S", "T", "W", "ANN", "ARG", "BLE", "COM", "DJ", "DTZ", "EM",
+    "ERA", "EXE", "FBT", "ICN", "INP", "ISC", "NPY", "PD", "PGH", "PIE", "PL", "PT", "PTH", "PYI", "RET", "RSE", "RUF",
+    "SIM", "SLF", "TCH", "TID", "TRY", "UP", "YTT"
+]
+unfixable = []
+
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "venv",
+]
+
+line-length = 120
+
+# Allow unused variables when underscore-prefixed.
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+
+# Assume Python 3.9
+target-version = "py39"
```

Comment:
Huh. This is fine, but do we want to do py38 since that's the trailing version we support? Or update to py311 when we support that?

Reply:
Sure, I'll update.